### PR TITLE
iosevka, iosevka-bin: 2.3.3 -> 3.2.2

### PR DESCRIPTION
**Pull request by rileyinman**
###### Motivation for this change
Iosevka has a new stable release! The build process has remained the same, but several options (such as Iosevka Term being renamed to Iosevka Fixed, as well as individual character style names) have been changed. This may break some custom builds, so I'm not sure if there should be a warning or if we should leave it up to users to update their build options.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/fonts/iosevka/package.json
Comment (asymmetric) [inline pkgs/data/fonts/iosevka/package.json:3]: Should we have 

```json
"engines": {
	"node": ">=12.16.0"}
```

here?

**Review by asymmetric — COMMENTED** [review]
I think [this](https://github.com/NixOS/nixpkgs/blob/37cdb3c5c96ed5b9b0c539e064466d379456889f/pkgs/development/node-packages/node-packages-v10.json#L92) should be moved to `node-packages-v12.json`, since Iosevka now [requires](https://github.com/be5invis/Iosevka/blob/09e506362da05b70c76c01da090542d15e210b77/package.json#L6) Node > 12.

For the same reason, we might need

```nix
  iosevka = callPackage ../data/fonts/iosevka {
    nodejs = nodejs-12_x;
    nodePackages = nodePackages_12_x;
  };
```

in `all-packages.nix`.

**Comment by rileyinman** [comment]
Good catch, I'll rebuild it after moving that.

File: pkgs/data/fonts/iosevka/package.json
Comment (rileyinman) [reply pkgs/data/fonts/iosevka/package.json:3]: I don't see why not!

Comment (asymmetric) [reply pkgs/data/fonts/iosevka/package.json:3]: I think I misunderstood the goal of this file. IIUC it's not there to be evaluated by NPM to build the package (upstream's `package.json` is used for that) - this is just here to know which packages need to be added to the `nodePackages` package set.

Sorry for the confusion, I think the addition I suggested is actually not needed.

**Comment by prusnak** [comment]
Please rework your PR. It now has a merge conflict after merging https://github.com/NixOS/nixpkgs/pull/89184

**Comment by rileyinman** [comment]
@prusnak Rebased to master in order to fix. In the meantime a new update came out, so this no longer updates to 3.0.1 but rather 3.1.0.

File: pkgs/data/fonts/iosevka/bin.nix
Comment (asymmetric) [inline pkgs/data/fonts/iosevka/bin.nix:15]: hash mismatch in fixed-output derivation '/nix/store/kzdqlrvvcvk15ddzg7jlgwyvwmp2qlpy-iosevka-bin-3.1.0':
  wanted: sha256:1hpwf4nyjl8rxwjiwmva07c0mn0pqyydxnzsvrzavydjivfqykzk
  got:    sha256:1mnf6yp0v1svkhjf598yhcbihhd150nxrddk14d84ardk0d6z3na

Comment (rileyinman) [reply pkgs/data/fonts/iosevka/bin.nix:15]: Oh, I ran `nix-prefetch-url` incorrectly. Thanks for the catch!

**Comment by asymmetric** [comment]
@rileyinman could you squash your commits following the guidelines?

**Comment by rileyinman** [comment]
Commits squashed, in the meantime a bugfix release was made so the PR now updates to 3.1.1.

**Comment by rileyinman** [comment]
Rebased to master to remove node conflict.

**Comment by asymmetric** [comment]
@rileyinman thanks for squashing :) 

AFAIK though, each package should be its own commit. [Here](https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches) are the guidelines - they dont' mention this explicitly, so I guess it's up for interpretation, but that's how I would do it.

**Comment by rileyinman** [comment]
Done, PR split into one for each package.

**Comment by asymmetric** [comment]
FYI I didn't mean one PR for each package, but one *commit*.

**Comment by rileyinman** [comment]
Whoops :sweat_smile:

**Comment by asymmetric** [comment]
BTW, why do we need a binary package in the first place?

**Comment by rileyinman** [comment]
According to the initial commit of the bin.nix, it was added to support platforms that can't build otfcc.

**Comment by rileyinman** [comment]
Rebased to master to fix node-packages conflict.

**Review by asymmetric — COMMENTED** [review]
I haven't tried this because it's trying to build something massive (node?), but LGTM.

Thanks for your work @rileyinman :)

**Comment by AluisioASG** [comment]
Since this is still open, can you pass `--jCmd=$NIX_BUILD_CORES` after `npm build --` so that it uses the right amount of cores?

**Comment by rileyinman** [comment]
@Alan01252 Done—I also updated it to the latest version and rebased to master to fix the merge conflict with node-packages.

**Review by asymmetric — CHANGES_REQUESTED** [review]
The `zsh_history` file shouldn't have been created, I reckon.

**Comment by asymmetric** [comment]
@AluisioASG could you point me to the documentation of the `--jCmd` flag?

**Comment by rileyinman** [comment]
No idea how that got in there, good catch :P

**Comment by AluisioASG** [comment]
> @AluisioASG could you point me to the documentation of the `--jCmd` flag?

@asymmetric There isn't AFAIK, Iosevka's build system (verda) doesn't even have a source code link. I had to download the package from NPM and look around for a way to control the number of jobs (it defaults to the number of cores in the system and was overwhelming my VM).

**Comment by asymmetric** [comment]
@rileyinman could you rebase this please? It's a bit unfortunate this hasn't gotten a review from the maintainers yet - a friendly ping :)

**Comment by rileyinman** [comment]
@asymmetric Funny story, you just caught me right as I was doing that!

**Review by ttuegel — APPROVED** [review]

**Comment by rileyinman** [comment]
Thank you!

**Comment by asymmetric** [comment]
Is it just me or are the Etoile and Aile variants not being built?

**Pull request merged**
